src/tally.rs

```diff
@@ -0,0 +1,265 @@
+
+    let mut set = JoinSet::new();
+    for day in 1..=25
+    {
+        let path = path.clone();
+        set.spawn(async move {
+            let path = day_path(path, day).await;
+            (path, day)
+        });
+    }
+
+    let mut have = Vec::new();
+    let mut dont_have = Vec::new();
+    while let Some(Ok(res)) = set.join_next().await
+    {
+        // The day path exists
+        match res
+        {
+            (Ok(path), day) => have.push((path, day)),
+            (_, day) => dont_have.push(day),
+        }
+    }
+
+
+    Ok((have, dont_have))
+}
+
+async fn build_day(day: u32) -> Result<(), AocError>
+{
+    let bin = format!("day_{:02}", day);
+    let res = tokio::process::Command::new("cargo")
+        .args(["build", "--release", "--bin", &bin])
+        .output()
+        .await?;
+
+    if !res.status.success()
+    {
+        Err(AocError::BuildError(bin.to_string()))
+    }
+    else
+    {
+        Ok(())
+    }
+}
+
+async fn build_days(cargo_folder: PathBuf, days: &[(PathBuf, u32)]) -> Result<Vec<u32>, AocError>
+{
+    let mut set = JoinSet::new();
+    for (_path, day) in days.iter()
+    {
+        let day = *day;
+        set.spawn(async move { build_day(day).await });
+    }
+
+    while let Some(Ok(res)) = set.join_next().await
+    {
+        res?;
+    }
+
+    let mut vec = Vec::new();
+    for (pb, day) in days
+    {
+        let pb = pb.clone();
+        let day = *day;
+        if run_day(cargo_folder.clone(), (pb, day), 1).await.is_err()
+        {
+            vec.push(day);
+        }
+    }
+
+    Ok(vec)
+}
+
+fn parse_get_times(output: Output) -> Result<(usize, Option<usize>), AocError>
+{
+    let parse = |line: &str| -> Result<usize, AocError> {
+        let start = line.find('(').ok_or_else(|| AocError::ParseStdout)?;
+        let stop = line.find("ms)").ok_or_else(|| AocError::ParseStdout)?;
+        Ok(line[start + 1..stop].parse().unwrap())
+    };
+    let text = std::str::from_utf8(&output.stdout).unwrap();
+    let mut iter = text.split('\n');
+    let p1 = parse(iter.next().unwrap())?;
+    let p2 = iter.next().and_then(|n| parse(n).ok());
+
+    Ok((p1, p2))
+}
+
+async fn run_day(
+    cargo_folder: PathBuf,
+    day: (PathBuf, u32),
+    number_of_runs: usize,
+) -> Result<(usize, Option<usize>), AocError>
+{
+    let bin = format!("day_{:02}", day.1);
+    let mut target = cargo_folder;
+    target.push("target/release");
+    target.push(&bin);
+
+    let mut set = JoinSet::new();
+    for _ in 0..number_of_runs
+    {
+        let dir = day.0.clone();
+        let target = target.clone();
+        set.spawn(
+            async move { tokio::process::Command::new(target).current_dir(dir).output().await },
+        );
+    }
+    let mut vec: Vec<(usize, Option<usize>)> = Vec::new();
+    while let Some(Ok(res)) = set.join_next().await
+    {
+        if let Err(ref e) = res
+        {
+            // Just try again if we have too many files open
+            if e.raw_os_error()
```

Comment:
Seems a bit hacky, but it's better if it works than it to error on that 🤷

Reply:
yeah